src/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs

```diff
@@ -264,4 +269,191 @@ public sealed partial class RefreshPropertiesAttribute : System.Attribute
         public override bool IsDefaultAttribute() { throw null; }
     }
 }
-
+namespace System.ComponentModel.Design
+{
+    public partial interface IDesignerHost : System.ComponentModel.Design.IServiceContainer, System.IServiceProvider
```

Comment:
In other words if https://github.com/dotnet/corefx/blob/d8fc81080723d58ca569c891ac1ebd86119b6f88/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs#L648 is the only place it is used then lets remove that usage and not move all this design stuff down into primitives.